### PR TITLE
add emailverificationservice to verify email by code

### PR DESCRIPTION
**Pull request by noodlemoodle**
nano https://github.com/nanoPayinc/NANOPAY/pull/15391

### CODE REVIEW

File: src/foam/nanos/auth/email/EmailVerificationCode.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+  properties: [
+    {
+      class: 'String',
+      name: 'verificationCode',
+      required: true,
+      view: function(_, X) {
+        return X.data.FragmentedTextField.create({
+          delegates: [
+            X.data.FragmentedTextFieldFragment.create({ maxLength: 1 }, X),
```
Comment (jlhughes) [inline src/foam/nanos/auth/email/EmailVerificationCode.js:45]: Why the dashes? Who would enter it this way? or is this to explicitly show that there are four digits to enter?

File: src/foam/nanos/auth/email/ClientEmailVerificationService.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
```
Comment (kgrgreer) [inline src/foam/nanos/auth/email/ClientEmailVerificationService.js]: This isn't needed anymore. Just add:
  client: true,

in EmailVerificationService.

File: src/foam/nanos/auth/email/EmailVerificationService.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  documentation: `
+    Email verification service to verify user email by code.
+  `,
+  methods: [
+    {
+      name: 'verifyByCode',
+      type: 'Void',
+      async: true,
+      args: [
```
Comment (kgrgreer) [inline src/foam/nanos/auth/email/EmailVerificationService.js]: You could just do:
  args: 'Context x, String email'

File: src/foam/nanos/auth/email/EmailVerificationService.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        { name: 'x', type: 'Context' },
+        { name: 'email', type: 'String' }
+      ]
+    },
+    {
+      name: 'verifyCode',
+      type: 'Boolean',
+      async: true,
+      args: [
```
Comment (kgrgreer) [inline src/foam/nanos/auth/email/EmailVerificationService.js]: and here

File: src/foam/nanos/auth/email/ServerEmailVerificationService.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+  messages: [
+    { name: 'RESEND_MESSAGE', message: 'This code is no longer valid, a new code has been sent to your email address.'}
+  ],
+
+  methods: [
+    {
+      name: 'verifyByCode',
+      type: 'Void',
+      args: [
```
Comment (kgrgreer) [inline src/foam/nanos/auth/email/ServerEmailVerificationService.js]: And here and elsewhere.

File: src/foam/nanos/auth/email/VerificationCodeView.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+ foam.CLASS({
+  package: 'foam.nanos.auth.email',
+  name: 'VerificationCodeView',
+  extends: 'foam.u2.View',
+
+  documentation: 'view to enter email verification code',
+
+  imports: [
+    'ctrl',
```
Comment (kgrgreer) [inline src/foam/nanos/auth/email/VerificationCodeView.js]: Where is this used?

File: src/foam/nanos/auth/email/EmailVerificationCode.js (hunk 1 above)
Comment (noodlemoodle) [reply src/foam/nanos/auth/email/EmailVerificationCode.js:45]: Screen Shot 2022-10-03 at 1 32 19 PM

yes this is just for the appearance, the user doesnt actually have to enter the dashes

**Pull request merged**